react-basic-practice/src/pages/ProductDetail/index.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { getProductById } from '@/services/Products';
+import { updateCart, getCartByUserId } from '@/services/Cart';
+import DefaultLayout from '@/layouts/DefaultLayout';
+import Quantity from '@/components/Quantity';
+import Button from '@/components/common/Button';
+import Popup from '@/components/Popup';
+
+import { useLoading } from '@/contexts/LoadingProvider';
+import { useToast } from '@/contexts/ToastProvider';
+
+import './index.css';
+
+const ProductDetail = () => {
+  const { productId } = useParams();
+  const { loading, setLoading } = useLoading();
+  const [product, setProduct] = useState([]);
+  const [quantityProduct, setQuantity] = useState(0);
+  const {
+    openPopup, setOpenPopup, status, setStatus, message, setMessage,
+  } = useToast();
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        setLoading(true);
+        const data = await getProductById(productId);
+        data ? setProduct(data) : setProduct([]);
+      } catch {
+        alert('Error loading data, please reload the page');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    getData();
+  }, []);
+
+  const handleSetQuantity = (value) => {
+    setQuantity(value);
+  };
+
+  const handleAddCart = async () => {
+    let productAddCart = [];
+    try {
+      const dataCart = await getCartByUserId(1);
```

Comment:
Why `1`?

Reply:
Because I don't have a specific login user id, I'm trying to get the data of the user with id 1 to show data